fieldservice_route_availability/models/fsm_order.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2025 APSL-Nagarro Antoni Marroig
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class FSMRoute(models.Model):
+    _inherit = "fsm.order"
+
+    @api.constrains("fsm_route_id", "scheduled_date_start")
+    def check_black_out_days(self):
+        for order in self:
+            if order.fsm_route_id and order.scheduled_date_start:
+                for group in order.fsm_route_id.fsm_blackout_group_ids:
+                    match = group.fsm_blackout_day_ids.filtered(
+                        lambda x, order=order: x.date
+                        == order.scheduled_date_start.date()
+                    )
+                    if match:
+                        raise ValidationError(
+                            _(
+                                "The day %(date)s is a blackout day in the route"
+                                " of the field service operations"
```

Comment:
I noticed the phrasing in the validation error message could be improved to sound more natural, something like:

```suggestion
                                "The date %(date)s is a blackout day for field"
                                " service operations on this route."
```

Reply:
Thanks for the comments, done!